_posts/2019-04-06-omakase-relay.markdown

```diff
@@ -0,0 +1,240 @@
+drastically reduces the need for style in a Relay component.
+
+Co-location's biggest selling point is simplicity, having everything you need in one place makes it easier to
+understand how a component works. This makes code review simpler, and lowers the barrier to understanding the
+entire systems at scale.
+
+## Community
+
+When we adopted Relay, there was no competition - we'd have just used the fetch API. Over time, Apollo came up and
+really put a considerable amount of effort into lowering the barriers to entry, and making it feasible to build
+complex apps easily.
+
+We did an audit last year of what it would take to re-create a lot of the infrastructure we use in Relay atop of
+the (much more popular) Apollo GraphQL eco-system and saw it was feasible but would require a considerable amount
+of work across many different plugins and tools. With Relay that's all packaged into one tool, works consistently
+and obviously doesn't have a scaling problem as Facebook have tens of thousands of Relay components.
+
+It's worth highlighting the core difference in community management for Apollo vs Relay. Engineers working on
+Apollo have great incentives to do user support, and improve the tools for the community - that's their businesses
+value. Relay on the other hand is used in many places at Facebook, and the engineers on the team support internal
+issues first. IMO, this is reasonable, Relay is an opinionated batteries-included framework for building user
+interfaces, and ensuring it works with the baffling amount of JavaScript at Facebook is more or less all the team
+has time for.
+
+That leaves space for the OSS community to own their own problems.
+
+## Scale Safety
+
+Relay puts a lot of emphasis on ahead of time safety. The Relay compiler validates your queries against your
+GraphQL schema, we extended the compiler to create TypeScript types for the composed API fragments and there are
+strict naming systems enforced by the compiler. All of these guides engineers to build scalable codebases.
+
+How this works in practice is that whenever you need to change the data a component requires, you edit the
+fragment, the Relay compiler verifies your query, if successful then your TypeScript types are updated and you can
+use the new property in your React component above. See below for a [quick video][vid] showing the Relay compiler
+in action:
+
+{% include epic_video.html url="/images/omakase-relay/relay-process-720.mov" title="Relay isolation tree" style="width:100%; display:block;" %}
+
+## Cultural Fit
+
+Relay fit well into our team because:
+
+- We had engineers who were interested in contributing back and making it work for our cases, this is much less of
+  an issue now that Relay has matured and is better documented.
+- We had engineers used to using ahead-of-time error validation tools like compilers
+- We saw a lot of value in a tightly coupling our view structure to our user interface
```

Comment:
```suggestion
- We saw a lot of value in a tightly coupling our view structure to our user interface.
```

Not sure I really understand what this is trying to convey. View structure on disk?